### ISSUE TITLE
Shell test failed in conformance testing

### ISSUE THREAD

**Issue by TomSweeneyRedHat** [containers/buildah#953]
During conformance testing it was discovered:

That the shell test failed.

buildah bud t buildahimage .

buildah inspect buildah-image

Dockerfile contents

```

FROM centos:7

SHELL ["/bin/bash", "-xc"]

RUN env

```

Per Yiqiao Pu

[1] https://github.com/ypu/buildah/tree/conformance

Buildah v1.4

**Comment by rhatdan** [comment]
What exactly failed?

**Comment by rhatdan** [comment]
I think this is working correctly,  buildah by default builds OCI Images and Docker is building docker images.  If you want Shell to work, you need to build with --format docker

```
 buildah bud --format docker -t shell -f ~/Dockerfile.shell ~
STEP 1: FROM centos:7
STEP 2: SHELL ["/bin/bash", "-xc"]
STEP 3: RUN env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/
SHLVL=1
HOME=/root
_=/usr/bin/env
STEP 4: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev]localhost/shell:latest
Getting image source signatures
Skipping fetch of repeat blob sha256:1d31b5806ba40b5f67bde96f18a181668348934a44c9253b420d5f04cfb4e37a
Copying blob sha256:e598f918c029ce150735d8e0fa8e24545a972680d9a8d272c15adf09268c7bb3
 159 B / 159 B [============================================================] 0s
Copying config sha256:15e7a274a6c65decf84f6a94b1137617f52bd4912d6025b00188bd4b42784589
 2.30 KiB / 2.30 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
--> 15e7a274a6c65decf84f6a94b1137617f52bd4912d6025b00188bd4b42784589

#buildah inspect --format '{{.Docker.Config.Shell}}' shell
[/bin/bash -xc]
```

**Comment by rhatdan** [comment]
@ypu Could you rerun the tests with format set to docker 

export BUILDAH_FORMAT=docker

Will force this for all buildah/podman commands.

**Comment by TomSweeneyRedHat** [comment]
As @rhatdan noted, setting the environment variable gets us at least closer to the expected results.

**Comment by rhatdan** [comment]
What is still broken here?

**Comment by TomSweeneyRedHat** [comment]
I don't think anything is still broken here with the format change that you suggested.  We probably could close this but I'd like a head nod from @ypu

**Comment by ypu** [comment]
Hi @TomSweeneyRedHat and @rhatdan 
I test with export "BUILDAH_FORMAT=docker" and "--format docker" to the command line directly. Seems the problem still exist. The "SHELL" is set in "ContainerConfig" but not effect "CMD".

Here is the difference for "CMD" in "ContainerConfig":
From buildah image:
`
        "ContainerConfig": {
...
            "Cmd": [
                "/bin/bash"
            ],
...
            "Shell": [
                "/bin/bash",
                "-xc"
            ]
        },
`
From docker image:
`
        "ContainerConfig": {
...
            "Cmd": [
                "/bin/bash",
                "-xc",
                "env"
            ],
 ...
            "Shell": [
                "/bin/bash",
                "-xc"
            ]
        },
...
`

I test with upstream code at Aug 24(HEAD: 9a4425427583fb70031e4b69bea7a81a8cbdbacc).

**Comment by rhatdan** [comment]
I take it that this is fixed in buildah 1.5
